README.md

```diff
@@ -31,20 +25,122 @@ Download
 Download [the latest JAR][2] or grab via Maven:
 ```xml
 <dependency>
-  <groupId>com.squareup</groupId>
+  <groupId>com.squareup.tape2</groupId>
   <artifactId>tape</artifactId>
-  <version>1.2.3</version>
+  <version>2.0.0-SNAPSHOT</version>
 </dependency>
 ```
 or Gradle:
 ```groovy
-compile 'com.squareup:tape:1.2.3'
+compile 'com.squareup.tape2:tape:2.0.0-SNAPSHOT'
 ```
 
 Snapshots of the development version are available in [Sonatype's `snapshots` repository][snap].
 
 
 
+Usage
+--------
+
+Create a `QueueFile` instance.
+
+```java
+File folder = context.getDir("app", Context.MODE_PRIVATE);
+File file = new File(folder, "queue");
+QueueFile queueFile = new QueueFile.Builder(file).build();
+```
+
+Add some data to the queue to the end of the queue. `QueueFile` accepts a `byte[]` of arbitrary length.
+
+```java
+queueFile.add("data".getBytes());
+```
+
+Read data at the head of the queue.
+
+```java
+byte[] data = queueFile.peek();
+```
+
+Remove processed elements.
+
+```java
+// Remove the eldest element.
+queueFile.remove();
+
+// Remove multiple elements.
+queueFile.remove(n);
+
+// Remove all elements.
+queueFile.clear();
+```
+
+Read and process multiple elements with the iterator API.
+
+```java
+Iterator<byte[]> iterator = queueFile.iterator();
+while (iterator.hasNext()) {
+  byte[] element = iterator.next();
+  process(element);
+  iterator.remove();
+}
+```
+
+While `QueueFile` works with `byte[]`, `ObjectQueue` works with arbitrary Java objects with a similar API. An `ObjectQueue` may be backed by a persistent `QueueFile`, or in memory. A persistent `ObjectQueue` requires a `Converter` to encode and decode objects.
+
+```java
+// A persistent ObjectQueue.
+ObjectQueue<String> queue = ObjectQueue.create(queueFile, converter);
```

Comment:
I'll make that change.

For next steps, it might be valuable to just have a few party converter implementations (Moshi + Gson) as seperate artifacts in the repo and link to any 3rd party ones.